### DISCUSSION TITLE
The lock file is not up to date with the latest changes in composer.json.

### DISCUSSION THREAD

**Discussion by ashidid** [composer/composer#10487] [Q&A / Support]
While working with composer we frequently get following warning,

Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.

Is there a mechanism or command which will give the details where exactly this mismatch is found? Finding this mismatch is very time consuming. Simply running `composer update --lock` may wrongly give an acceptance to false version.

**Comment by stof** [comment]
This lock file warning is only based on comparing the content-hash in the lock file to the hash computed for the composer.json file. So it can know that something changed, but it cannot know what has changed.

> Simply running `composer update --lock` may wrongly give an acceptance to false version.

What do you mean here ?

**Reply by Seldaek** [reply]
Yeah the only way to avoid this is to always run updates when you change something, so you don't end up in an outdated state. If you do get an outdated state, update --lock is an okish way to get out of it, it will still enforce that the depencies in the composer.json are valid with the lock file (and otherwise error asking you to allowlist packages that need updates) so you can't get completely wrong versions.
